dpypelines/pipeline/dataset_ingress_v1.py

```diff
@@ -34,25 +36,28 @@ def dataset_ingress_v1(files_dir: str, pipeline_config: dict):
     # Attempt to access the local data store
     try:
         local_store = LocalDirectoryStore(files_dir)
+        logger.info("Local data store successfully instansiated", data ={"local_store_dir": files_dir, "local_store": local_store})
     except Exception as err:
+        logger.error(f"Failed to access local data at {files_dir}", err, data={"local_store_dir": files_dir})
         de_notifier.failure()
-        raise Exception(
-            message.unexpected_error(f"Failed to access local data at {files_dir}", err)
-        ) from err
+
+        raise err
 
     # Extract the patterns for required files from the pipeline configuration
     try:
         required_file_patterns = matching.get_required_files_patterns(pipeline_config)
+        logger.info("Required file patterns retrieved from pipeline configuration", data={"required_file_patterns": required_file_patterns})
     except Exception as err:
+        logger.error("Failed to get required files pattern", err, data={"pipeline_config": pipeline_config, "local_store": local_store})
         de_notifier.failure()
-        raise Exception(
-            message.unexpected_error("Failed to get required files patterns", err)
-        ) from err
+
+        raise err
 
     # Check for the existence of each required file
     for required_file in required_file_patterns:
         try:
             if not local_store.has_lone_file_matching(required_file):
+                logger.info(f"Required matching file {required_file} was not found.", data={"required_file": required_file})
```

Comment:
I think you can get rid of most of this block. Id we dont' get a match just raise `FileNotFoundError(f"Could not find file matching pattern {required_file}")` and that will do. The error log after the catch will give us all the information we're after.